### DISCUSSION TITLE
Use `this` keyword to define constructor (saves duplication)

### DISCUSSION THREAD

**Discussion by greg-stein** [dotnet/csharplang#5760] [Language Ideas]
So, this feature is pretty simple. Instead of this:
```
public class SomeClassName
{
    public SomeClassName(int n) {}
    public SomeClassName(string s) {}
    public SomeCLassName() {}
}
```
write this:
```
public class SomeClassName
{
    public this(int n) {}
    public this(string s) {}
    public this() {}
}
```

This way you have to specify class name only once. Not to say, the constructors are visually distinguished from the rest of class methods very nicely. Just a proposal. What would you say?

**Comment by CyrusNajmabadi** [comment]
Dupe of https://github.com/dotnet/csharplang/discussions/1928

**Reply by greg-stein** [reply]
No this is not a dupe. But this one is: [#1335](https://github.com/dotnet/csharplang/issues/1335)

**Reply by CyrusNajmabadi** [reply]
it is a dupe.  THe discussion i linked to is about "Nam[ing] constructor with keyword instead of name", which is what this proposes as well.  IT's a discussion so that the different ideas can be put out there (e.g. do you use 'this' or 'new' or 'ctor' etc.).
